File: features/lender/src/main/java/com/nivasafinance/features/lender/lender/entity/Lender.java
package com.nivasafinance.features.lender.lender.entity;

import com.nivasafinance.common.audit.AuditableEntity;
import com.nivasafinance.features.lender.lender.enums.LenderStatus;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Table(name = "n_lender")
@NoArgsConstructor
@AllArgsConstructor
@Data
@EqualsAndHashCode(callSuper = true)
public class Lender extends AuditableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "key", nullable = false, length = 20)
    private String key;

    @Column(name = "name", nullable = false, length = 100)
    private String name;

    @Column(name = "status", nullable = false, length = 40)
    @Enumerated(EnumType.STRING)
    private LenderStatus status;
}

File: features/lender/src/main/java/com/nivasafinance/features/lender/lender/entity/Lender.kt
package com.nivasafinance.features.lender.lender.entity

import com.nivasafinance.common.audit.AuditableEntity
import com.nivasafinance.features.lender.lender.enum.LenderStatus
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.util.UUID

@Entity
@Table(name = "n_lender")
data class Lender(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: UUID? = null,

    @Column(name = "key", nullable = false, length = 20)
    val key: String,

    @Column(name = "name", nullable = false, length = 100)
    val name: String,

    @Column(name = "status", nullable = false, length = 40)
    @Enumerated(EnumType.STRING)
    val status: LenderStatus

) : AuditableEntity()
